numpy/ma/core.py

```diff
@@ -4509,48 +4516,41 @@ def all(self, axis=None, out=None):
         """
         mask = _check_mask_axis(self._mask, axis)
         if out is None:
-            d = self.filled(True).all(axis=axis).view(type(self))
+            d = self.filled(True).all(axis=axis, **kwargs).view(type(self))
             if d.ndim:
                 d.__setmask__(mask)
-            elif mask:
+            elif mask and self.size != 0:
```

Comment:
Tiny but to me either of the following would read slightly more clearly:
```
elif mask and self.size:
    ...
```
or
```
elif mask and self.size > 0:
    ...
```
(i.e., thinking of it as "the object has a mask and size"; of course, size cannot be negative).